### ISSUE TITLE
Integration with Github

### ISSUE THREAD

**Issue by gkarugi** [nanobox-io/nanobox#640]
How can I deploy directly from a GitHub repository, say master, after a commit.
I have tried checking for documentation on this but found nothing.

**Comment by philipgiuliani** [comment]
To do that, you need a continious integration system which builds your image. I don't know which CI platforms are supported, I'm personally using [Gitlab CI](https://about.gitlab.com/features/gitlab-ci-cd/), but I know that [Travis CI](https://travis-ci.org) is also supported.

There is the following documentation: https://docs.nanobox.io/workflow/ci-cd/

**Comment by dthompson86** [comment]
When you setup your GitLab CI, did you find it necessary to self host? I'm looking at trying to use Nanobox with a GitLab setup and don't know if it's better to be self hosted or to host with them. There don't seem to be any examples walking through a GitLabCI + Nanobox setup.

**Comment by LeCoupa** [comment]
Yes the lack of documentation for Gitlab CI is a real pain. I see this issue on a regular basis on the community slack.

**Comment by danhunsaker** [comment]
GitLab CI cloud doesn't support test runners with the required access to Docker; for deeply technical reasons, Docker Remote doesn't work for Nanobox.

As to not having docs on it, that's probably largely because nobody has offered to submit any? See nanobox-io/nanobox-docs for that.
